 tcc.h    | 13 +++++++++++++
 tccelf.c |  4 ++--
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/tcc.h b/tcc.h
index 9702464..13ef188 100644
--- a/tcc.h
+++ b/tcc.h
@@ -39,6 +39,12 @@
 #include <setjmp.h>
 #include <time.h>
 
+#ifdef __VSF__
+# undef _WIN32
+# undef _WIN64
+# undef _MSC_VER
+#endif
+
 #ifndef _WIN32
 # include <unistd.h>
 # include <sys/time.h>
@@ -197,6 +203,13 @@ extern long double strtold (const char *__nptr, char **__endptr);
 # endif
 #endif
 
+#ifdef __VSF__
+# undef TCC_IS_NATIVE
+# if defined __VSF__ && defined __WIN__
+#  define TCC_TARGET_PE
+# endif
+#endif
+
 #if defined CONFIG_TCC_BACKTRACE && CONFIG_TCC_BACKTRACE==0
 # undef CONFIG_TCC_BACKTRACE
 #else
diff --git a/tccelf.c b/tccelf.c
index 15ed168..a3833dd 100644
--- a/tccelf.c
+++ b/tccelf.c
@@ -2286,8 +2286,8 @@ static int layout_sections(TCCState *s1, int *sec_order, struct dyn_inf *d)
 		   headers itself (and the ELF header as well), it'll
 		   come out with same memory use but will make various
 		   tools like binutils strip work better.  */
-		ph->p_offset = 0;
-		ph->p_vaddr = base;
+//		ph->p_offset = 0;
+//		ph->p_vaddr = base;
             }
             ph->p_paddr = ph->p_vaddr;
             ++n;