 lib/urldata.h                     | 4 ++--
 src/tool_operate.c                | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/urldata.h b/lib/urldata.h
index cc9c88870..552c66134 100644
--- a/lib/urldata.h
+++ b/lib/urldata.h
@@ -172,8 +172,8 @@ typedef CURLcode (*Curl_datastream)(struct Curl_easy *data,
    libcurl able to do significantly faster uploads in some circumstances. Even
    larger buffers can help further, but this is deemed a fair memory/speed
    compromise. */
-#define UPLOADBUFFER_DEFAULT 65536
-#define UPLOADBUFFER_MAX (2*1024*1024)
+#define UPLOADBUFFER_DEFAULT 16384
+#define UPLOADBUFFER_MAX (16384)
 #define UPLOADBUFFER_MIN CURL_MAX_WRITE_SIZE
 
 #define CURLEASY_MAGIC_NUMBER 0xc0dedbadU
diff --git a/src/tool_operate.c b/src/tool_operate.c
index 5f4b76b0b..01e90fae5 100644
--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -205,7 +205,7 @@ static curl_off_t VmsSpecialSize(const char *name,
 }
 #endif /* __VMS */
 
-#define BUFFER_SIZE (100*1024)
+#define BUFFER_SIZE (32*1024)
 
 #ifdef __VSF__
 #   define transfersl               (curl_tool_operate_ctx->transfersl)
